ERCS/erc-7432.md

```diff
@@ -323,94 +287,85 @@ The following JSON is an example of [ERC-7432](./eip-7432.md) Metadata:
 ```
 
 The `roles` array properties are SUGGESTED, and developers should add any other relevant information as necessary (e.g.,
-an image for the role). However, it's highly RECOMMENDED to include the `isUniqueRole` property, as this field is
-used to determine if the `hasRole` or `hasNonUniqueRole` function should be called (refer to 
-[Unique and Non-Unique Roles](#unique-and-non-unique-roles)).
-
-It's also important to highlight the importance of the `inputs` property. This field describes the parameters that
-should be encoded and passed to the `grantRole` function. It's RECOMMENDED to use the properties `type` and `components`
-defined on the Solidity ABI Specification, where `type` is the canonical type of the parameter, and `components` is used
-for complex tuple types.
+an image for the role). It's also important to highlight the importance of the `inputs` property. This field describes
+the parameters that should be encoded and passed to the `grantRole` function. It's RECOMMENDED to use the properties
+`type` and `components` defined on the Solidity ABI Specification, where `type` is the canonical type of the parameter,
+and `components` is used for complex tuple types.
 
 ### Caveats
 
 * Compliant contracts MUST implement the `IERC7432` interface.
-* A role is represented by a `bytes32`, and it's RECOMMENDED to use the `keccak256` of the role's name for this purpose:
-  `bytes32 role = keccak256("ROLE_NAME")`.
-* The `grantRoleFrom` and `grantRevocableRoleFrom` functions MUST revert if the `_expirationDate` is in the past or if
-  the `msg.sender` is not approved to grant roles on behalf of the `_grantor`. It MAY be implemented as `public` or
- `external`.
-* The `revokeRole` and `revokeRoleFrom` functions SHOULD revert if `_revocable` is `false`. They MAY be implemented as `public` or `external`.
-* The `revokeRoleFrom` function MUST revert if the `msg.sender` is not approved to revoke roles on behalf of the `_revoker` or the `_grantee`.
-  It MAY be implemented as `public` or `external`.
+* A role is represented by a `bytes32`, and it's RECOMMENDED to use the `keccak256` of the role's name with its inputs:
+  `bytes32 roleId = keccak256("RoleName(input_type)")`.
+* The `grantRole` function MUST revert if the `expirationDate` is in the past or if the `msg.sender` is not approved to
+  grant roles on behalf of the NFT owner. It MAY be implemented as `public` or `external`.
```

Comment:
**issue (blocking):**  It's important to specify that the `grantRole` function also MUST lock (deposit or freeze) the token.